discord/user.py

```diff
@@ -100,13 +101,19 @@ def __ne__(self, other):
     def __hash__(self):
         return self.id >> 22
 
+    def has_flag(self, flag: UserFlags):
+        v = flag.value
+        return (self._public_flags & v) == v
+
     def _update(self, data):
         self.name = data['username']
         self.id = int(data['id'])
         self.discriminator = data['discriminator']
         self.avatar = data['avatar']
+        self._public_flags = data.get("public_flags", 0)
         self.bot = data.get('bot', False)
         self.system = data.get('system', False)
+        self.public_flags = PublicFlags(self)
```

Comment:
This should be a property. See e.g. `Guild.system_message_flags` or `Message.flags` for an example of what I mean.